futurex_openedx_extensions/helpers/export_data.py

```diff
@@ -0,0 +1,179 @@
+        query_string = urlencode(query_params)
+        full_url = f'{self.request.scheme}://{self.request.get_host()}{self.request.path}'
+        if query_string:
+            full_url += f'?{query_string}'
+        return full_url
+
+    def get_filtered_query_params(self) -> dict:
+        """Filter query params - to avoid infinite requests loop"""
+        query_params = self.request.GET.copy()
+        query_params.pop('download', None)
+        query_params.pop('filename', None)
+        return query_params
+
+    def generate_csv_url_response(self) -> dict:
+        """Return response with csv file url"""
+        response_data = {'download_url': '', 'error': ''}
+        filtered_query_params = self.get_filtered_query_params()
+        view_url = self.get_view_request_url(filtered_query_params)
+        view_data = {
+            'user_id': self.request.user.id,
+            'view_class_name': self.__class__.__name__,
+            'query_params': filtered_query_params,
+            'kwargs': self.kwargs,
+        }
+
+        error = None
+        try:
+            csv_file, error = export_data_to_csv(
+                view_url, view_data, self.request.fx_permission_info, self.filename
+            )
+            response_data['download_url'] = self.build_download_url(csv_file)
+        except Exception as exc:  # pylint: disable=broad-exception-caught
+            response_data['error'] = f'Failed to write CSV file due to internal error: {str(exc)}'
+
+        if error:
+            response_data['error'] = error
+
+        return response_data
+
+    def list(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        """Override the list method to generate CSV and return JSON response with CSV URL"""
+        if self.request.query_params.get('download') == 'csv':
+            response = self.generate_csv_url_response()
+            return Response(response, status=http_status.HTTP_200_OK)
+
+        return super().list(request, args, kwargs)
+
+
+def _get_mocked_request(url: str, user: Any, fx_info: dict, query_params: dict) -> Request:
+    """Create mocked request"""
+    factory = APIRequestFactory()
+    mocked_request = factory.get(url)
+    mocked_request.user = user
+    mocked_request.fx_permission_info = fx_info
+    mocked_request.query_params = query_params
+    return mocked_request
+
+
+def _get_user(user_id: int) -> Any:
+    """Get User from user_id"""
+    if user_id:
+        try:
+            return User.objects.get(id=user_id)
+        except User.DoesNotExist:
+            # Error handling need to be done
+            pass
+    return None
+
+
+def _get_view_class_instance(view_name: str) -> Any:
+    """Create view class instance from given iew name"""
+    view_instance = None
+    dasboard_views_module = importlib.import_module('futurex_openedx_extensions.dashboard.views')
+    view_class = getattr(dasboard_views_module, view_name, None)
```

Comment:
please take a look at the [function here](https://github.com/nelc/futurex-openedx-extensions/blob/a06200515e70dc4302e8efc5f20c5cf7f1e8c846/tests/test_dashboard/test_views.py#L57), which is a bit cleaner. I believe we can use `resolve`instead of `importlib`